src/Message/ApplePayAuthorizeRequest.php

```diff
@@ -0,0 +1,386 @@
+    }
+
+    /**
+     * Sets the merchant ID sent with the request
+     *
+     * @param string $value
+     * @return static
+     */
+    public function setMerchantID($value)
+    {
+        return $this->setParameter('merchantSessionIdentifier', $value);
+    }
+
+    /**
+     * @return null|string
+     */
+    public function getDomainName()
+    {
+        return $this->getParameter('domainName');
+    }
+
+    /**
+     * Sets the domain name sent with the request
+     *
+     * @param string $value
+     * @return static
+     */
+    public function setDomainName($value)
+    {
+        return $this->setParameter('domainName', $value);
+    }
+
+    /**
+     * @return null|string
+     */
+    public function getDisplayName()
+    {
+        return $this->getParameter('displayName');
+    }
+
+    /**
+     * Sets the display name sent with the request
+     *
+     * @param string $value
+     * @return static
+     */
+    public function setDisplayName($value)
+    {
+        return $this->setParameter('displayName', $value);
+    }
+
+    /**
+     * @return null|string
+     */
+    public function getSignatureID()
+    {
+        return $this->getParameter('signature');
+    }
+
+    /**
+     * Sets the signature for the request
+     *
+     * @param string $value
+     * @return static
+     */
+    public function setSignatureID($value)
+    {
+        return $this->setParameter('signature', $value);
+    }
+
+    /**
+     * @return null|string
+     */
+    public function getUsername()
+    {
+        return $this->getParameter('username');
+    }
+
+    /**
+     * Sets the username
+     *
+     * @param string $value
+     * @return static
+     */
+    public function setUsername($value)
+    {
+        return $this->setParameter('username', $value);
+    }
+
+    /**
+     * @return null|string
+     */
+    public function getPassword()
+    {
+        return $this->getParameter('password');
+    }
+
+    /**
+     * Sets the password
+     *
+     * @param string $value
+     * @return static
+     */
+    public function setPassword($value)
+    {
+        return $this->setParameter('password', $value);
+    }
+
+    /**
+     * Overriding AbstractReuqest::sendData() so that we can make a REST call instead of a SOAP call.
+     * @param array $data
+     * @return Response
+     */
+    public function sendData($data)
+    {
+        $httpRequest  = $this->createClientRequest($data);
+        $httpResponse = $httpRequest->send();
+
+        $statusCodes = array(
+            'message' => $httpResponse->getReasonPhrase(),
+            'statusCode' => $httpResponse->getStatusCode()
+        );
+
+        try {
+            $message = $httpResponse->json();
+            $response = array_merge(
+                $statusCodes,
+                $message
+            );
+        // If you try to parse an empty response body, error will be thrown.
+        } catch (\RunTimeException $e) {
```

Comment:
is it possible to detect the empty response body before calling `json()` rather than doing exception handling?